woodwork/statistics_utils/_get_medcouple_statistic.py

```diff
@@ -0,0 +1,47 @@
+import numpy as np
+
+
+def _calculate_medcouple_statistic(y):
+    """Calculates the medcouple statistic. Based on the implementation by statsmodels and off of the paper
+    M. Hubert and E. Vandervieren, "An adjusted boxplot for skewed distributions" Computational Statistics
+    & Data Analysis, vol. 52, pp. 5186-5201, August 2008.
+
+    Args:
+        y (np.ndarray, pd.Series): Data on which sampling will occur.
+
+    Returns:
+        float: Medcouple statistic
+    """
+    y = np.squeeze(np.asarray(y))
+    y = np.sort(y)
+
+    n = y.shape[0]
+    if n % 2 == 0:
+        mf = (y[n // 2 - 1] + y[n // 2]) / 2
+    else:
+        mf = y[(n - 1) // 2]
+
+    z = y - mf
```

Comment:
nit, but would be really helpful if these variables had more helpful names. Would make it easier to understand in the future